subxt/src/transaction.rs

```diff
@@ -389,14 +389,17 @@ impl<'client, T: Config, E: Decode + HasModuleError, Evs: Decode>
             let ev = ev?;
             if &ev.pallet == "System" && &ev.variant == "ExtrinsicFailed" {
                 let dispatch_error = E::decode(&mut &*ev.data)?;
-                if let Some((pallet_idx, error_idx)) =
-                    dispatch_error.module_error_indices()
-                {
-                    let details = self.client.metadata().error(pallet_idx, error_idx)?;
+                if let Some(raw_error) = dispatch_error.module_error_indices() {
+                    // Error index is utilized as the first byte from the error array.
+                    let details = self
+                        .client
+                        .metadata()
+                        .error(raw_error.pallet_index, raw_error.error[0])?;
```

Comment:
It would be better to encapsulate `raw_error.error[0])` in a method e.g. `raw_error.error_index()`